lib/rubocop/cop/style/ternary_parentheses.rb

```diff
@@ -34,57 +34,103 @@ module Style
       #   foo = (bar && baz) ? a : b
+      #
+      # @example
+      #
+      #   EnforcedStyle: require_parentheses_when_complex
+      #
+      #   @bad
+      #   foo = (bar?) ? a : b
+      #   foo = (bar.baz?) ? a : b
+      #   foo = bar && baz ? a : b
+      #
+      #   @good
+      #   foo = bar? ? a : b
+      #   foo = bar.baz ? a : b
+      #   foo = (bar && baz) ? a : b
       class TernaryParentheses < Cop
         include IfNode
         include SafeAssignment
         include ConfigurableEnforcedStyle
 
         MSG = '%s parentheses for ternary conditions.'.freeze
+        MSG_COMPLEX = 'Use parentheses %sfor ternary conditions with' \
+          ' complex expressions.'.freeze
 
         def on_if(node)
-          return unless ternary?(node)
-
-          add_offense(node, node.source_range, message) if offense?(node)
+          return unless ternary?(node) && !infinite_loop? && offense?(node)
+          add_offense(node, node.source_range, message(node))
         end
 
         private
 
         def offense?(node)
           condition, = *node
 
-          (require_parentheses? && !parenthesized?(condition) ||
-            !require_parentheses? && parenthesized?(condition)) &&
-            !(safe_assignment?(condition) && safe_assignment_allowed?) &&
-            !infinite_loop?
+          if safe_assignment?(condition)
+            !safe_assignment_allowed?
+          else
+            parens = parenthesized?(condition)
+            case style
+            when :require_parentheses_when_complex
+              condition_complex?(condition) ? !parens : parens
+            else
+              require_parentheses? ? !parens : parens
+            end
+          end
         end
 
         def autocorrect(node)
           condition, = *node
 
-          return nil if !require_parentheses? && (safe_assignment?(condition) ||
+          return nil if parenthesized?(condition) &&
+                        (safe_assignment?(condition) ||
                         unsafe_autocorrect?(condition))
 
           lambda do |corrector|
-            if require_parentheses?
-              corrector.insert_before(condition.source_range, '(')
-              corrector.insert_after(condition.source_range, ')')
-            else
+            if parenthesized?(condition)
               corrector.remove(condition.loc.begin)
               corrector.remove(condition.loc.end)
+            else
+              corrector.insert_before(condition.source_range, '(')
+              corrector.insert_after(condition.source_range, ')')
             end
           end
         end
 
-        def message
-          verb = require_parentheses? ? 'Use' : 'Omit'
+        def condition_complex?(condition)
+          if condition.type == :begin
+            condition.to_a.any? { |x| condition_complex?(x) }
+          else
+            non_complex_type?(condition) ? false : true
+          end
+        end
+
+        def non_complex_type?(condition)
```

Comment:
Btw, documenting this method and the preceding method might useful for the people reading this code in the future. :-)